test/subtype.jl

```diff
@@ -765,6 +772,16 @@ function test_intersection()
 
     @testintersect(DataType, Type, DataType)
     @testintersect(DataType, Type{T} where T<:Integer, Type{T} where T<:Integer)
+
+    @testintersect((Type{Tuple{Vararg{T}}} where T), Type{Tuple}, Bottom)
+    @testintersect(Tuple{Type{S}, Tuple{Any, Vararg{Any}}} where S<:Tuple{Any, Vararg{Any}},
+                   Tuple{Type{T}, T} where T,
+                   Tuple{Type{S},S} where S<:Tuple{Any,Vararg{Any,N} where N})
+
+    @test_broken isequal_type(_type_intersect(Tuple{T,T} where T,
+                                              Union{Tuple{S,Array{Int64,1}},Tuple{S,Array{S,1}}} where S),
+                              Union{Tuple{Vector{Int64},Vector{Int64}},
+                                    Tuple{Vector{T},Vector{T}} where T>:Vector})
```

Comment:
just dropping by to note some other examples of the same test here, but that don't involve Union:
```julia
julia> A = Tuple{T, T} where T;

julia> B = Tuple{Array{S, 1}, S} where S;

julia> _type_intersect(A, B)
Tuple{T, T} where T <: (Array{S, 1} where S)

julia> _type_intersect(B, A)
Tuple{Array{S, 1}, Array{S, 1}} where S

julia> B = Tuple{S, Array{S, 1}} where S;

julia> _type_intersect(A, B)
Tuple{Array{Any, 1}, Array{Any, 1}}

julia> _type_intersect(B, A)
Tuple{Array{Any, 1}, Array{Any, 1}}
```